### ISSUE TITLE
Python test framework reports a crash when a server is killed by the test framework

### ISSUE THREAD

**Issue by Tryneus** [rethinkdb/rethinkdb#4976]
This isn't a huge problem as it doesn't stop the test, but when the table fuzzer kills a server while we are still waiting to read its ports from the log, it may print:

```
Unhandled exception in thread started by <bound method Process.read_ports_from_log of <driver.Process object at 0x7fd9e44a6c10>>
Traceback (most recent call last):
  File "/home/ssd2/grey/rethinkdb/test/common/driver.py", line 727, in read_ports_from_log
    self.check()
  File "/home/ssd2/grey/rethinkdb/test/common/driver.py", line 764, in check
    raise RuntimeError("%s %s stopped unexpectedly with return code: %r" % (self.server_type.capitalize(), self._name, self.returncode))
RuntimeError: Server a stopped unexpectedly with return code: -9
```

This gets a little spammy and makes it harder to visually parse the output.

**Comment by larkost** [comment]
A fix for this us up for review with CR 3516 from branch `larkost/4976-quiet-read_ports_from_log-error`

**Comment by larkost** [comment]
This has been merged into ~next` with a488d4d5.
